### ISSUE TITLE
regarding sft fine-tuning

### ISSUE THREAD

**Issue by sorobedio** [BunsenFeng/model_swarm#3]
hello, thank you for the great work.
i would like to have the code you use to fine-tune on validation set for the sft experiment. i want to try the same on gemma3. the goal is better understand the tuning process then reproducing the exact results.
i am refering to the results presented in Table 9(SFT)
thank you.
